app/src/main/java/com/example/d3companion/presenters/list/ListPresenter.kt

```diff
@@ -1,21 +1,27 @@
 package com.example.d3companion.presenters.list
 
+import android.util.Log
 import com.example.d3companion.models.D3ViewElement
 import com.example.d3companion.models.D3ViewType
+import com.example.d3companion.services.D3IcyVeinsListener
+import com.example.d3companion.services.FileD3IcyVeinsProvider
 import com.example.d3companion.views.list.ListView
 import java.lang.ref.WeakReference
 import java.util.ArrayList
 
-class ListPresenter(view: ListView) : ListPresenterProvider {
+class ListPresenter(view: ListView, provider: FileD3IcyVeinsProvider) : ListPresenterProvider, D3IcyVeinsListener {
 
     private var weakView: WeakReference<ListView>? = null
+    private val dataProvider: FileD3IcyVeinsProvider = provider
 
     private val items: MutableList<D3ViewElement> = ArrayList()
 
     init {
         weakView = WeakReference(view)
+        dataProvider.setListener(this)
```

Comment:
This method will likely be changed to `dataProvider.listener = this`, if you follow the comment on the class header.